core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -534,6 +549,14 @@ public boolean hasLicensesXml() {
      *             thrown if one or several mandatory dependencies doesn't exists.
      */
     /*package*/ void resolvePluginDependencies() throws IOException {
+        String requiredCoreVersion = getRequiredCoreVersion();
+        if (requiredCoreVersion == null) {
+            throw new IOException(shortName + " doesn't declare required core version.");
+        } else {
+            if (Jenkins.getVersion().isOlderThan(new VersionNumber(requiredCoreVersion))) {
```

Comment:
Should be made conditional on `ENABLE_PLUGIN_DEPENDENCIES_VERSION_CHECK` I suppose.